File: app/src/main/java/org/ametro/ui/navigation/entities/NavigationHeader.java
package org.ametro.ui.navigation.entities;

import android.graphics.drawable.Drawable;

public class NavigationHeader extends NavigationItem {

    private final Drawable icon;
    private final String city;
    private final String country;
    private final String comment;
    private final Drawable[] transportTypes;


    public NavigationHeader(Drawable icon, String city, String country, String comment, Drawable[] transportTypes) {
        this.icon = icon;
        this.city = city;
        this.country = country;
        this.comment = comment;
        this.transportTypes = transportTypes;
    }

    public NavigationHeader(String emptyText) {
        this(null, emptyText, null, null, new Drawable[0]);
    }

    public Drawable getIcon() {
        return icon;
    }

    public String getCity() {
        return city;
    }

    public String getCountry() {
        return country;
    }

    public String getComment() {
        return comment;
    }

    public Drawable[] getTransportTypeIcons() {
        return transportTypes;
    }
}

File: app/src/main/java/io/github/romangolovanov/apps/ametro/ui/navigation/entities/NavigationHeader.kt
package io.github.romangolovanov.apps.ametro.ui.navigation.entities

import android.graphics.drawable.Drawable

class NavigationHeader(
    val icon: Drawable?,
    val city: String?,
    val country: String?,
    val comment: String?,
    private val transportTypes: Array<Drawable?>
) : NavigationItem() {

    constructor(emptyText: String) : this(null, emptyText, null, null, emptyArray())

    fun getTransportTypeIcons(): Array<Drawable?> = transportTypes
}
